taskcluster/translations_taskgraph/util/dataclass_helpers.py

```diff
@@ -0,0 +1,426 @@
+"""
```

Comment:
This section is probably the most complicated part of this PR, as it's walking the types and handling some custom logic. This top comment should hopefully explain what's going on.